### PR TITLE
support fixed-wings in new Gazebo(gz or ignition gazebo)

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
### Solved Problem
Currently, the new gazebo does not support fixed-wing / VTOL type vehicles due to the missing interface for the lift drag plugin

As we want to support the new Gazebo moving forward, this is one of the major features that is missing before we can switch to the new gazebo.

Fixes https://github.com/PX4/PX4-Autopilot/issues/20836

### Solution
I took the plane model in https://github.com/PX4/PX4-Autopilot/tree/pr-gz-bridge-airspeed from @ahcorde and modified it so that the joint angles can be controlled via the px4 `gz_bridge`

What was done:
- Added a airframe config for the plane model
- Modified the plane model to have correct thrust directions
- Added a joint controller to take care of the control surface deflection
- Added a servo control publication in `gz_bridge` to publish joint position commands for the control surfaces. Joint names need to strictly follow a naming convention `px4_servo_<i>` depending on the index

### Test coverage
To test,
```
make px4_sitl gz_plane
```
Video of first flight: https://youtu.be/4heMHqKiYUI
[First Flight](https://youtu.be/4heMHqKiYUI "First Flight")

### Still Unresolved issues
- [ ] Tune Flight Controls
- [ ] The servo outputs are non-dimensional, but the control surface deflections are in joint angles.
- [ ] It is still unclear what the development workflow would look like if someone is to develop on gazebo plugins. Current workflow does not include custom plugins
- [ ] Add airspeed sensor 
  - I believe @ahcorde is trying to address in https://github.com/PX4/PX4-Autopilot/issues/20836

### CODE REVIEW

**Comment by bperseghetti** [comment]
Nice, I've worked on adding functionality to the joint position controller system to control groups as well as use subtopics that inherit the model name. I also made the Ackermann steering plugin behave the way we want it to allowing for phi (center steering angle published as a double)

**Comment by Jaeyoung-Lim** [comment]
> Nice, I've worked on adding functionality to the joint position controller system to control groups as well as use subtopics that inherit the model name. I also made the Ackermann steering plugin behave the way we want it to allowing for phi (center steering angle published as a double)

Great! Is this somewhere I can use? or are you saying that this is already what I am using?

**Comment by bperseghetti** [comment]
> > Nice, I've worked on adding functionality to the joint position controller system to control groups as well as use subtopics that inherit the model name. I also made the Ackermann steering plugin behave the way we want it to allowing for phi (center steering angle published as a double)
> 
> Great! Is this somewhere I can use? or are you saying that this is already what I am using?
> 
> > adding functionality to the joint position controller system to control groups
> 
> I would like to start separating motors(velocity control) and servos (joint control), but I think we need to make the distinction on the gz_bridge with control allocation first

Yep, that's what I'm working on. Haha 
Here is what we needed for Ackermann: https://github.com/gazebosim/gz-sim/pull/1860
Here is my work for multiple joint_name and sub_topic for joint-controller and joint-position-controller:
https://github.com/rudislabs/gz-sim/tree/pr-multi-joint-with-subtopic

I'll finish implementing the joint-controller later today, but the joint-position-controller works perfect.

**Comment by bperseghetti** [comment]
@Jaeyoung-Lim @dagar Here is what will allow us to do multi-vehicle like what I enabled for the multicopter:
https://github.com/gazebosim/gz-sim/pull/1861
https://github.com/gazebosim/gz-sim/pull/1860

**Comment by ahcorde** [comment]
When I execute 

```bash
make px4_sitl gz_x500
```

I got this error:

```bash
INFO  [dataman] data manager file './dataman' size is 7866640 bytes
INFO  [init] starting gazebo with world: /tmp/PX4-Autopilot/Tools/simulation/gz/worlds/default.sdf
WARN  [init] PX4_GZ_MODEL_NAME or PX4_GZ_MODEL not set using PX4_SIM_MODEL.
INFO  [gz_bridge] world: default, model name: plane_0, simulation model: plane
libEGL warning: DRI2: failed to create dri screen
libEGL warning: DRI2: failed to create dri screen
ERROR [gz_bridge] Service call timed out
ERROR [gz_bridge] Task start failed (-1)
ERROR [init] gz_bridge failed to start
ERROR [px4] Startup script returned with return value: 256
```

File: Tools/simulation/gz/models/plane/model.sdf [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+          <damping>1.000</damping>
+        </dynamics>
+      </axis>
+      <physics>
+        <ode>
+          <implicit_spring_damper>1</implicit_spring_damper>
+        </ode>
+      </physics>
+    </joint>
+    <joint name='rudder_joint' type='revolute'>
```
Comment (ahcorde) [inline Tools/simulation/gz/models/plane/model.sdf:434]:
```suggestion
    <joint name='px4_servo_4' type='revolute'>
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by ahcorde** [comment]
I merged with `main`, but I removed the merge and it's working. thank you

File: Tools/simulation/gz/models/plane/model.sdf (hunk 1 above)
Comment (Jaeyoung-Lim) [reply Tools/simulation/gz/models/plane/model.sdf:434]: Thanks!

**Review by dagar — APPROVED** [review]

**Comment by dagar** [comment]
Needs more work (tuning, Fortress support questions, etc), but this should be good enough to merge into main so we can continue iterating.

**Pull request merged**